fuzzamoto-scenarios/bin/ir.rs

```diff
@@ -207,8 +207,30 @@ where
     }
 
     fn process_actions(&mut self, actions: Vec<CompiledAction>) {
+        let handshake_opts = fuzzamoto::connections::HandshakeOpts {
```

Comment:
I think we should extend the IR to include the `HandshakeOpts`. So in the end it'd be nice for new connection to either do the handshake with options specified in the IR or no handshake at all.

e.g. a `LoadHandshakeOpts(...)` and an additional `AddConnectionAndHandshake` operation which takes the opts and a time variable as input.